module/zfs/qat_compress.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+#ifndef	_SYS_QAT_COMPRESS_H
+#define	_SYS_QAT_COMPRESS_H
+
+#if defined(_KERNEL) && defined(HAVE_QAT)
+#include <sys/zio.h>
+#include "cpa.h"
+#include "dc/cpa_dc.h"
+extern int qat_init(void);
+extern void qat_fini(void);
+extern int use_qat(size_t s_len);
+extern int qat_compress(int dir, char *src, int src_len,
```

Comment:
Please refine the first parameter as follows:

```
enum qat_direction {
    QAT_COMPRESS = 0,
    QAT_DECOMPRESS
};
```

And then call it as ```qat_compress(QAT_COMPRESS, ...)``` or ```qat_compress(QAT_DECOMPRESS, ...)```. Or even better if you can rename ```qat_compress()``` to ```qat_call()``` so that it becomes less confusion.